plugins/kubernetes-backend/src/service/KubernetesProxy.test.ts

```diff
@@ -657,29 +637,137 @@ describe('KubernetesProxy', () => {
         },
       ),
     );
-    permissionApi.authorize.mockResolvedValue([
-      { result: AuthorizeResult.ALLOW },
-    ]);
+
     clusterSupplier.getClusters.mockResolvedValue([
       {
         name: 'cluster1',
         url: 'http://localhost:9999/subpath',
         authProvider: '',
       },
     ]);
+
     authTranslator.decorateClusterDetailsWithAuth.mockImplementation(
       async x => x,
     );
-    const app = express().use(
-      Router().use('/mountpath', proxy.createRequestHandler({ permissionApi })),
-    );
 
-    const requestPromise = request(app)
-      .get('/mountpath/api/v1/namespaces')
-      .set(HEADER_KUBERNETES_CLUSTER, 'cluster1');
-    worker.use(rest.all(requestPromise.url, (req: any) => req.passthrough()));
+    const requestPromise = setupProxyPromise({
+      proxyPath: '/mountpath',
+      requestPath: '/api/v1/namespaces',
+
+      headers: {
+        [HEADER_KUBERNETES_CLUSTER]: 'cluster1',
+      },
+    });
+
     const response = await requestPromise;
 
     expect(response.status).toEqual(200);
   });
+
+  describe('WebSocket', () => {
+    const proxyPath = '/proxy';
+    const wsPath = '/ws';
+
+    let wsPort: number;
+    let proxyPort: number;
+    let wsEchoServer: WebSocketServer;
+    let expressServer: Server;
+
+    const eventPromiseFactory = (
+      ws: WebSocket,
+      event: 'connection' | 'open' | 'close' | 'error' | 'message',
+    ) => new Promise(resolve => ws.once(event, x => resolve(x?.toString())));
+
+    beforeAll(async () => {
+      await new Promise(resolve => {
+        expressServer = express()
+          .use(
+            Router()
+              .use(proxyPath, proxy.createRequestHandler({ permissionApi }))
+              .use(errorHandler()),
+          )
+          .listen(0, '0.0.0.0', () => {
+            proxyPort = (expressServer.address() as AddressInfo).port;
+            resolve(null);
+          });
+      });
+
+      wsEchoServer = new WebSocketServer({
+        // server: expressServer,
```

Comment:
Maybe forgot to delete?
```suggestion
```

Reply:
Indeed, testing if using the same server would work, was not giving me a good result.